src/main/java/com/dabsquared/gitlabjenkins/connection/GitLabConnectionProperty.java

```diff
@@ -153,8 +156,9 @@ public FormValidation doTestConnection(
             item.checkPermission(Item.CONFIGURE);
             try {
                 GitLabConnection gitLabConnectionTested = null;
-                GitLabConnectionConfig descriptor =
-                        (GitLabConnectionConfig) Jenkins.getInstance().getDescriptor(GitLabConnectionConfig.class);
+                GitLabConnectionConfig descriptor = (GitLabConnectionConfig)
+                        Objects.requireNonNull(Jenkins.getInstance()).getDescriptor(GitLabConnectionConfig.class);
+                assert descriptor != null;
```

Comment:
Again, same as above: Can we use the if-else statement instead of assert here?

Reply:
Certainly, I'd be happy to provide more context on why I chose to use assert instead of an if-else statement in that particular scenario.

1. Intent:
   - assert statements are primarily used for debugging and catching programming errors during development. They express assumptions about the state of the program.

2. Preconditions and Fail Fast:
   - I used assert to check critical preconditions – conditions that are crucial for the correct functioning of the code. This aligns with the "fail fast" principle, helping us detect and address errors as early as possible in the development process.

3. Reducing Cognitive Load:
   - Using assert for debugging-related checks separates assumptions from regular control flow logic, making the code more readable. It also emphasizes the significance of critical conditions.

Reply:
These code snippets are not for testing purposes, but is used instead for program flow control. Can you please update these?

Reply:
ok sure